src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -40,2428 +40,1339 @@
+            size = numKeysDist(Catch::rng());
+            for (auto const& e :
+                 LedgerTestUtils::generateValidUniqueLedgerEntriesWithTypes(
+                     {CONTRACT_DATA, CONTRACT_CODE}, size))
+            {
+                resources.footprint.readWrite.emplace_back(LedgerEntryKey(e));
+            }
 
-        // Recreation of persistent entries should fail.
-        putWithFootprint("key", 0, contractKeys, {lk}, 1000,
-                         /*expectSuccess*/ false,
-                         ContractDataDurability::PERSISTENT);
+            auto tx =
+                test.createInvokeTx(resources, scFunc, scArgs, 100,
+                                    std::numeric_limits<uint32_t>::max() - 100);
 
-        // Extend operation should skip expired entries
-        extendOp(1'000, {lk});
+            // Only read entry, write entry, and TX size fees
+            auto expectedFee = test.getTxSizeFees(tx) +
+                               test.getEntryReadFee(resources) +
+                               test.getEntryWriteFee(resources);
+            checkFees(expectedFee, tx);
+        }
+    }
 
-        // Make sure liveUntilLedger is unchanged by extendOp
-        checkContractDataLiveUntilLedger(
-            "key", ContractDataDurability::PERSISTENT, initLiveUntilLedger);
+    uniform_int_distribution<uint32_t> bytesDist(1, 100'000);
+    SECTION("readBytes fee")
+    {
+        resources.readBytes = bytesDist(Catch::rng());
 
-        // Restore the entry
-        restoreOp({lk}, 20048);
+        auto tx =
+            test.createInvokeTx(resources, scFunc, scArgs, 100,
+                                std::numeric_limits<uint32_t>::max() - 100);
 
-        ledgerSeq = getLedgerSeq(*app);
-        checkContractDataArchivalState(
-            "key", ContractDataDurability::PERSISTENT, ledgerSeq, true);
+        // Only readBytes and TX size fees
+        auto expectedFee =
+            test.getTxSizeFees(tx) + test.getReadBytesFee(resources);
+        checkFees(expectedFee, tx);
+    }
 
-        // Entry is considered to be restored on lclNum (as we didn't close an
-        // additional ledger).
-        checkContractDataLiveUntilLedger(
-            "key", ContractDataDurability::PERSISTENT,
-            ledgerSeq + stateArchivalSettings.minPersistentTTL - 1);
+    SECTION("writeBytes fee")
+    {
+        resources.writeBytes = bytesDist(Catch::rng());
+        auto tx =
+            test.createInvokeTx(resources, scFunc, scArgs, 100,
+                                std::numeric_limits<uint32_t>::max() - 100);
 
-        // Write to entry to check that is is live
-        put("key", 1, ContractDataDurability::PERSISTENT);
+        // Only writeBytes and TX size fees
+        auto expectedFee =
+            test.getTxSizeFees(tx) + test.getWriteBytesFee(resources);
+        checkFees(expectedFee, tx);
     }
+}
 
-    SECTION("re-create expired temporary entry")
```

Comment:
What happened to this test? Does it exist somewhere else now?

Reply:
This was redundant, we check temp entry recreation in "TTL enforcement", `InvokeHostFunctionTests.cpp:1658`